apps/web/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -118,19 +121,25 @@ export function ProductDetailForm<F extends Field>({
               required
               value={quantityControl.value}
             />
-            <SubmitButton>{ctaLabel}</SubmitButton>
+            <SubmitButton disabled={disabled}>{ctaLabel}</SubmitButton>
           </div>
         </div>
       </form>
     </FormProvider>
   );
 }
 
-function SubmitButton({ children }: { children: React.ReactNode }) {
+function SubmitButton({ children, disabled }: { children: React.ReactNode; disabled?: boolean }) {
   const { pending } = useFormStatus();
 
   return (
-    <Button className="w-auto @xl:w-56" loading={pending} size="medium" type="submit">
+    <Button
+      className="w-auto @xl:w-56"
+      disabled={disabled}
```

Comment:
This should probably be disabled while `pending`. Does the `loading` prop not take care of that?